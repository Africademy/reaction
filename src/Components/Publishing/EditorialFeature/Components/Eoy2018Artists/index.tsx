import { color, Serif } from "@artsy/palette"
import { unica } from "Assets/Fonts"
import { ArticleProps } from "Components/Publishing/Layouts/FeatureLayout"
import { Nav } from "Components/Publishing/Nav/Nav"
import { ImageSetPreview } from "Components/Publishing/Sections/ImageSetPreview"
import React from "react"
import styled from "styled-components"
import { Eoy2018ArticleHeader } from "./ArticleHeader"

export const BORDER_WIDTH = 4

export class Eoy2018Artists extends React.Component<ArticleProps> {
  sectionIsHeader = section => {
    const isText = section.type === "text"
    const isHeader = isText && section.body.includes("<h1>")
    return isHeader
  }

  sectionIsHeaderImage = (section, i) => {
    const {
      article: { sections },
    } = this.props
    const isImage = section.type === "image_collection"
    const sectionBefore = i !== 0 && sections[i - 1]
    const sectionBeforeIsHeader = this.sectionIsHeader(sectionBefore)

    return isImage && sectionBeforeIsHeader
  }

  sectionHeader = (section, i) => {
    const {
      article: { sections },
    } = this.props
    const sectionAfter = sections[i + 1]
    const sectionAfterisImage = this.sectionIsHeaderImage(sectionAfter, i + 1)
    const src = sectionAfterisImage && sectionAfter.images[0].url

    return (
      <ArtistHeaderSection key={i}>
        <ArtistHeaderTitle dangerouslySetInnerHTML={{ __html: section.body }} />
        <ArtistHeaderImg src={src} />
      </ArtistHeaderSection>
    )
  }

  sectionText = (section, i) => {
    return (
      <TextSection size="5" key={i}>
        <div dangerouslySetInnerHTML={{ __html: section.body }} />
      </TextSection>
    )
  }

  sectionImageSet = (section, i) => {
    return (
      <ImageSetWrapper key={i}>
        <ImageSetPreview section={section} />
      </ImageSetWrapper>
    )
  }

  getSections = () => {
    const {
      article: { sections },
    } = this.props

    const renderedSections = sections.map((section, i) => {
      const isHeader = this.sectionIsHeader(section)
      const isHeaderImage = this.sectionIsHeaderImage(section, i)

      if (isHeader) {
        return this.sectionHeader(section, i)
      } else if (isHeaderImage) {
        return null
      } else {
        if (section.type === "text") {
          return this.sectionText(section, i)
        } else if (section.type === "image_set") {
          return this.sectionImageSet(section, i)
        } else {
          return null
        }
      }
    })
    return renderedSections
  }

  render() {
    return (
      <React.Fragment>
        <Nav canFix />

        <ArticleWrapper>
          <Eoy2018ArticleHeader />

          <ArticleContent>{this.getSections()}</ArticleContent>
        </ArticleWrapper>
      </React.Fragment>
    )
  }
}

const ArticleWrapper = styled.div`
  padding: 0 40px;
`

const ArticleContent = styled.div`
  padding: 40px 0;
  border-left: ${BORDER_WIDTH}px solid ${color("purple100")};

  blockquote {
    ${unica("s34")};
  }
`

const ArtistHeaderSection = styled.div`
  height: 55vh;
  border: ${BORDER_WIDTH}px solid ${color("purple100")};
  border-left: none;
  display: flex;
  margin-bottom: 40px;
`

const ArtistHeaderTitle = styled.div`
  flex: 1;
  border-right: ${BORDER_WIDTH}px solid ${color("purple100")};

  h1 {
    ${unica("s65")};
    width: 100%;
    height: 50%;
    padding: 20px;
    border-bottom: ${BORDER_WIDTH}px solid ${color("purple100")};
  }

  h2 {
    ${unica("s25")};
    width: 50%;
    height: 50%;
    display: inline-flex;
    padding: 20px;
    &:last-child {
      border-left: ${BORDER_WIDTH}px solid ${color("purple100")};
    }
    &:hover {
      background: ${color("purple100")};
      color: white;
    }
  }
`

const ArtistHeaderImg = styled.div<{ src?: string }>`
  flex: 1;
  ${props =>
    props.src &&
    `
    background: url(${props.src});
    background-size: cover;
    background-position: center;
  `};
`

const TextSection = styled(Serif)`
  max-width: 75%;
  margin-left: auto;
  padding-bottom: 60px;

  p {
    font-size: 24;
  }
`
const ImageSetWrapper = styled.div`
  margin-bottom: 60px;
  border: ${BORDER_WIDTH}px solid ${color("purple100")};
  border-left: none;
`
