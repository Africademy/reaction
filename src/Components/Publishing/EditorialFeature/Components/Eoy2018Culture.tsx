import { Box, Flex, Sans } from "@artsy/palette"
import { unica } from "Assets/Fonts"
import { Author, StyledAuthor } from "Components/Publishing/Byline/Author"
import { Date, DateContainer } from "Components/Publishing/Byline/Date"
import { Share } from "Components/Publishing/Byline/Share"
import { getArticleFullHref } from "Components/Publishing/Constants"
import {
  ArticleProps,
  FeatureLayout,
} from "Components/Publishing/Layouts/FeatureLayout"
import { Nav, NavContainer } from "Components/Publishing/Nav/Nav"
import React from "react"
import styled from "styled-components"

export const Eoy2018Culture: React.SFC<ArticleProps> = props => {
  const { article } = props
  return (
    <ArticleWrapper>
      <Nav canFix color="black" backgroundColor="white" />

      <ArticleHeader>
        <ArticleTitle size="16" weight="medium">
          <div>Year In Visual Culture</div>
          <div>2018</div>
        </ArticleTitle>
        <HeaderImg src={article.hero_section.url} />
      </ArticleHeader>

      <Intro>
        <Box width={"20%"}>
          <Share
            url={getArticleFullHref(article.slug)}
            title={article.social_title || article.thumbnail_title}
          />
        </Box>
        <Flex>
          <Author authors={article.authors} />
          <Date date={article.published_at} />
        </Flex>
      </Intro>

      <Box>
        <div dangerouslySetInnerHTML={{ __html: article.sections[0].body }} />
      </Box>
      <FeatureLayout {...props} />
    </ArticleWrapper>
  )
}

const ArticleWrapper = styled.div`
  ${NavContainer} {
    border-bottom-width: 6px;
    border-top: 6px solid black;
  }

  blockquote {
    ${unica("s34")};
  }
`

const ArticleTitle = styled(Sans)`
  text-transform: uppercase;
  padding: 20px 40px;
  border-bottom: 6px solid;
`

const Intro = styled(Flex)`
  align-items: center;
  padding: 20px 40px;

  ${DateContainer} {
    margin-top: 0;
  }
  ${StyledAuthor} {
    margin-top: 0;
  }
`

const HeaderImg = styled.img`
  height: 100%;
  object-fit: cover;
  object-position: center;
  width: 100%;
`

const ArticleHeader = styled.div`
  height: calc(100vh - 46px);
  overflow: hidden;
  border-bottom: 6px solid black;
`
