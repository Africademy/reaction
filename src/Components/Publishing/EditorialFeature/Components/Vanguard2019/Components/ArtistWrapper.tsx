import { Box, color, Flex, FlexProps, Sans, Serif } from "@artsy/palette"
import { Share, ShareContainer } from "Components/Publishing/Byline/Share"
import { getFullEditorialHref } from "Components/Publishing/Constants"
import { Emerging } from "Components/Publishing/EditorialFeature/Components/Vanguard2019/Blobs/Emerging"
import { GettingTheirDue } from "Components/Publishing/EditorialFeature/Components/Vanguard2019/Blobs/GettingTheirDue"
import { NewlyEstablished } from "Components/Publishing/EditorialFeature/Components/Vanguard2019/Blobs/NewlyEstablished"
import ArticleWithFullScreen from "Components/Publishing/Layouts/ArticleWithFullScreen"
import { StyledArtworkCaption } from "Components/Publishing/Sections/ArtworkCaption"
import { CaptionContainer } from "Components/Publishing/Sections/Caption"
import { FullScreenProvider } from "Components/Publishing/Sections/FullscreenViewer/FullScreenProvider"
import {
  Sections,
  StyledSections,
} from "Components/Publishing/Sections/Sections"
import { StyledText } from "Components/Publishing/Sections/StyledText"
import { ArticleData } from "Components/Publishing/Typings"
import { random } from "lodash"
import React from "react"
import styled from "styled-components"
import { slugify } from "underscore.string"
import { Media } from "Utils/Responsive"

export interface SVGBackgroundProps extends FlexProps {
  url?: string
}
export interface TextProps extends FlexProps {
  isMobile?: boolean
  isExpanded?: boolean
}
export class VanguardArtistWrapper extends React.Component<
  {
    article: ArticleData
    isExpanded?: boolean
    section?: string
    isMobile: boolean
    onSlideshowStateChange?: (state: boolean) => void
  },
  {
    isExpanded: boolean
  }
> {
  state = {
    isExpanded: this.props.isExpanded || false,
  }
  artistWrapper = null

  componentDidMount() {
    const {
      article: { title },
    } = this.props
    const slug = slugify(title)

    if (window.location.pathname.includes(slug)) {
      this.onExpand()
    }
  }

  onExpand() {
    const { isExpanded } = this.state

    if (isExpanded) {
      window.scrollTo({
        behavior: "smooth",
        top: this.artistWrapper.getBoundingClientRect().top + window.scrollY,
      })
    }

    this.setState({ isExpanded: !isExpanded })
  }

  getRandomSVG(section) {
    /* Lodash random returns a whole integer between 0 and the max passed.
     * "Emerging" has 20 SVG possibilities while Newly Established
     * and "Getting Their Due" have 15, hence the default to 15
     * */
    switch (section) {
      case "emerging":
        return random(20)
      default:
        return random(15)
    }
  }

  getSVGBackground(index, section) {
    switch (section) {
      case "emerging":
        return Emerging(index)
      case "newly-established":
        return NewlyEstablished(index)
      case "getting-their-due":
        return GettingTheirDue(index)
    }
  }

  render() {
    const { article, section, isMobile, onSlideshowStateChange } = this.props
    const { hero_section, layout, slug, title } = article
    const { isExpanded } = this.state

    const background = this.getSVGBackground(
      this.getRandomSVG(section),
      section
    )
    const backgroundColor = isExpanded ? color("black100") : color("white100")

    return (
      <FullScreenProvider onSlideshowStateChange={onSlideshowStateChange}>
        <ArtistWrapper
          background={backgroundColor}
          pt={50}
          mb={isExpanded && 100}
          id={slugify(article.title)}
          ref={artistWrapper => (this.artistWrapper = artistWrapper)}
        >
          <BackgroundContainer backgroundColor={backgroundColor}>
            {background}
          </BackgroundContainer>
          <ArticleWithFullScreen article={article}>
            <ArtistContainer
              pb={4}
              maxWidth={["100vw", 1000]}
              px={4}
              mx="auto"
              isMobile={isMobile}
              isExpanded={isExpanded}
            >
              <Box textAlign="center" pb={30}>
                <Media greaterThanOrEqual="xl">
                  <ArtistTitle size="12" element="h3">
                    {title}
                  </ArtistTitle>
                </Media>

                <Media lessThan="xl">
                  <InvertedSerif
                    size={["10", "12", "12"]}
                    element="h3"
                    isMobile={isMobile}
                    isExpanded={isExpanded}
                  >
                    {title}
                  </InvertedSerif>
                </Media>

                <Box position="relative">
                  {hero_section && (
                    <InvertedSans
                      size="4"
                      weight="medium"
                      isMobile={isMobile}
                      isExpanded={isExpanded}
                    >
                      {hero_section.deck}
                    </InvertedSans>
                  )}
                  <Flex justifyContent={["center", "flex-start"]}>
                    <Box position={["relative", "absolute"]} top={0}>
                      <Share
                        // TODO: We may need to use custom urls for in-page routing
                        url={getFullEditorialHref(layout, slug)}
                        title={title}
                        color={color("white100")}
                      />
                    </Box>
                  </Flex>
                </Box>
              </Box>

              <Sections
                hideAds
                article={article}
                customWidth={920}
                isTruncatedAt={!isExpanded && 2}
              />

              <ReadMoreButton
                size="5"
                weight="medium"
                textAlign="center"
                onClick={this.onExpand.bind(this)}
                isMobile={isMobile}
                isExpanded={isExpanded}
              >
                {isExpanded ? "Close" : "Read More"}
                <Sans size="8">{isExpanded ? "\u2191" : "\u2193"}</Sans>
              </ReadMoreButton>
            </ArtistContainer>
          </ArticleWithFullScreen>
        </ArtistWrapper>
      </FullScreenProvider>
    )
  }
}

export const InvertedSerif = styled(Serif)<TextProps>`
  mix-blend-mode: ${p => (p.isMobile ? "normal" : "difference")};
  color: ${color("white100")};
  will-change: color;
`

export const InvertedSans = styled(Sans)<TextProps>`
  mix-blend-mode: ${p => (p.isMobile ? "normal" : "difference")};
  color: ${color("white100")};
  will-change: color;
`

const ArtistTitle = styled(InvertedSerif)`
  font-size: 95px;
  line-height: 1em;
`

const ReadMoreButton = styled(InvertedSans)<{ onClick: () => void }>`
  text-transform: uppercase;
  cursor: pointer;
  color: ${p =>
    p.isMobile && p.isExpanded
      ? color("white100")
      : p.isMobile && !p.isExpanded
      ? color("black100")
      : color("white100")};
`

const ArtistContainer = styled(Box)<TextProps>`
  position: relative;

  /* override feature text drop-caps */
  p:first-child::first-letter,
  .paragraph:first-child::first-letter {
    font-family: inherit;
    font-size: inherit;
    float: none;
    line-height: inherit;
    margin-right: 0;
    margin-top: 0;
    text-transform: none;
  }

  ${StyledSections} {
    margin-top: 0;
  }

  ${ShareContainer},
  ${StyledText} {
    mix-blend-mode: ${p => (p.isMobile ? "normal" : "difference")};
    will-change: color;
    color: ${p =>
      p.isMobile && p.isExpanded
        ? color("white100")
        : p.isMobile && !p.isExpanded
        ? color("black100")
        : color("white100")};
  }

  ${CaptionContainer} {
    p {
      mix-blend-mode: ${p => (p.isMobile ? "normal" : "difference")};
      will-change: color;
      color: ${p => (p.isMobile ? color("black100") : color("white100"))};
    }

    a {
      mix-blend-mode: ${p => (p.isMobile ? "normal" : "difference")};
      will-change: color;
      color: ${p =>
        p.isMobile && !p.isExpanded ? color("black100") : color("white100")};
      background-image: linear-gradient(
        to bottom,
        transparent 0,
        #fff 1px,
        transparent 0
      );
      background-size: 1.25px 4px;
      background-repeat: repeat-x;
      background-position: bottom;
    }
  }

  ${StyledArtworkCaption} {
    mix-blend-mode: ${p => (p.isMobile ? "normal" : "difference")};
    color: ${p => (p.isMobile ? color("black100") : color("white100"))};
    will-change: color;

    a {
      mix-blend-mode: ${p => (p.isMobile ? "normal" : "difference")};
      will-change: color;
      color: ${p =>
        p.isMobile && !p.isExpanded ? color("black100") : color("white100")};
    }
  }

  ${StyledText} {
    a {
      mix-blend-mode: ${p => (p.isMobile ? "normal" : "difference")};
      will-change: color;
      color: ${p =>
        p.isMobile && p.isExpanded
          ? color("white100")
          : p.isMobile && !p.isExpanded
          ? color("black100")
          : color("white100")};
      background-image: linear-gradient(
        to bottom,
        transparent 0,
        #fff 1px,
        transparent 0
      );
      background-size: 1.25px 4px;
      background-repeat: repeat-x;
      background-position: bottom;
    }
  }
`

const BackgroundContainer = styled(Box)<{ backgroundColor: string }>`
  height: 100%;
  position: absolute;
  top: 0;
  left: 0;
  right: 0;
  background: ${({ backgroundColor }) => backgroundColor};
`

const ArtistWrapper = styled(Flex)`
  flex-direction: column;
  position: relative;
  padding-bottom: 100px;
  min-height: 100vh;
  z-index: 0;
`
