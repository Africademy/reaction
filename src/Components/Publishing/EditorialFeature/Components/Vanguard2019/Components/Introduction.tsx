import { Box, Flex, Sans, Serif } from "@artsy/palette"
import { Byline, BylineContainer } from "Components/Publishing/Byline/Byline"
import { Text } from "Components/Publishing/Sections/Text"
import { ArticleData } from "Components/Publishing/Typings"
import React from "react"
import styled from "styled-components"
import { VanguardVideoBackground } from "./VideoBackground"

export const VanguardIntroduction: React.SFC<{
  article: ArticleData
}> = props => {
  const { description } = props.article.series

  return (
    <IntroContainer>
      <Box minHeight="calc(100vh - 50px)" mb={150} pt={50}>
        <VanguardVideoBackground {...props} />
        <HeaderText pt={70} size="8" textAlign="center">
          The Artsy
        </HeaderText>
      </Box>

      <Box mx="auto" maxWidth={980} px={4}>
        <Flex flexDirection="column" alignItems="center" pb={50}>
          <Title size="12" element="h1" textAlign="center" pb={1}>
            The Artists To Know Right Now
          </Title>
          <Byline {...props} />

          <Box textAlign="center" pt={50}>
            <Sans size="3t" weight="medium">
              Video Headers by Alex John Beck
            </Sans>
            <Sans size="3t" weight="medium">
              Video Editing by Nate DeYoung
            </Sans>
            <Sans size="3t" weight="medium">
              Interaction Design by Wax Studios
            </Sans>
          </Box>
        </Flex>

        <Box pb={12}>
          <Text layout="standard" html={description} width="800px" />
        </Box>
      </Box>
    </IntroContainer>
  )
}

const HeaderText = styled(Sans)`
  font-size: 100px;
`

const IntroContainer = styled(Box)`
  ${BylineContainer} {
    div::before {
      display: none;
    }
  }
`

const Title = styled(Serif)`
  text-transform: uppercase;
  font-size: 90px;
  line-height: 1em;
`
