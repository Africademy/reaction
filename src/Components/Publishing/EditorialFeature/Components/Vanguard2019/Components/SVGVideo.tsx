import React from "react"
import styled from "styled-components"

interface SVGVideoBackgroundProps {
  id?: string
  url?: string
}

export const SVGVideoBackground = props => (
  <Video autoPlay loop muted playsInline controls={false} src={props.url} />
)

const Video = styled.video<SVGVideoBackgroundProps>`
  width: 100%;
  height: 100%;
  max-height: 95vh;
  object-fit: cover;
  position: absolute;
  z-index: -1;
  clip-path: url(#clip-svg-intro);
`
