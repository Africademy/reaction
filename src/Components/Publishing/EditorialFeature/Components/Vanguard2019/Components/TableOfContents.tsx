import { Box, Sans, Serif } from "@artsy/palette"
import { ArticleData } from "Components/Publishing/Typings"
import { times } from "lodash"
import React from "react"
import styled from "styled-components"
import { slugify } from "underscore.string"
import {
  NewlyEstablished1,
  NewlyEstablished2,
  NewlyEstablished3,
} from "../Assets/NewlyEstablished"

type SubSeriesType = "emerging" | "newly-established" | "getting-their-due"

export const VanguardTableOfContents: React.SFC<{
  article: ArticleData
  onChangeSection: (slug: string) => void
}> = props => {
  const {
    article: { relatedArticles },
    onChangeSection,
  } = props

  return (
    <Box pb={100} textAlign="center" minHeight="70vh">
      <Sans size={["8", "12", "14", "16"]} mb={4}>
        Table of Contents
      </Sans>

      {relatedArticles &&
        relatedArticles.map((subSeries, i) => {
          const subSeriesSlug = slugify(subSeries.title) as SubSeriesType
          return (
            <SeriesContainer
              key={i}
              onClick={() => onChangeSection(subSeriesSlug)}
            >
              <SvgContainer>{TocSvgs(subSeriesSlug)}</SvgContainer>
              <Chapter>
                <Serif
                  size={["8", "10", "12", "12"]}
                  pb={2}
                  maxWidth={1000}
                  mx="auto"
                >
                  {times(i + 1, () => "I")}
                </Serif>
                <SeriesTitle
                  size={["8", "10", "12", "14"]}
                  pb={2}
                  maxWidth={1000}
                  mx="auto"
                >
                  {subSeries.title}
                </SeriesTitle>
              </Chapter>
            </SeriesContainer>
          )
        })}
    </Box>
  )
}

export const TocSvgs = (type: SubSeriesType) => {
  // TODO: Update with emerging/getting due SVGs
  switch (type) {
    case "emerging":
      return <NewlyEstablished2 />
    case "newly-established":
      return <NewlyEstablished1 />
    case "getting-their-due":
      return <NewlyEstablished3 />
  }
}

const Chapter = styled(Box)`
  position: relative;
`

const SvgContainer = styled(Box)`
  position: absolute;
  left: 0;
  top: -50%;
  right: 0;
  bottom: auto;
  display: none;
`

const SeriesTitle = styled(Sans)`
  text-transform: uppercase;
`

const SeriesContainer = styled(Box)`
  position: relative;

  &:hover {
    ${SvgContainer} {
      display: block;
    }

    ${Chapter} {
      color: white;
      cursor: pointer;
    }
  }
`
