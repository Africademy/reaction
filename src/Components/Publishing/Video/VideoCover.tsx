import React, { Component } from "react"
import { Col } from 'react-styled-flexboxgrid'
import styled, { StyledFunction } from "styled-components"
import { media } from "../../Helpers"
import { Fonts } from "../Fonts"
import { IconVideoPlay } from "../Icon/IconVideoPlay"
import { MaxRow } from "./Shared"
import { VideoInfoBlock } from "./VideoInfoBlock"

interface Props {
  media?: any
  seriesTitle?: string
  description?: string
  playVideo?: () => void
  hideCover?: boolean
}

interface CoverProps {
  hideCover?: boolean
}

interface CoverAssetProps {
  src?: string
}

export class VideoCover extends Component<Props, null> {
  render() {
    const {
      description,
      media,
      seriesTitle,
      playVideo,
      hideCover
    } = this.props

    return (
      <VideoCoverContainer hideCover={hideCover}>
        <VideoCoverAsset src={media.cover_image_url} />
        <VideoCoverOverlay />
        <VideoCoverInfo>
          <VideoCoverInfoRow>
            <Col xs={2} sm={1} onClick={playVideo}>
              <IconVideoPlay color="white" />
            </Col>
            <Col xs={10} sm={3}>
              <VideoInfoBlock
                media={media}
                seriesTitle={seriesTitle}
              />
            </Col>
            <Col xs={12} sm={8}>
              <MediaDescription>
                {description}
              </MediaDescription>
            </Col>
          </VideoCoverInfoRow>
        </VideoCoverInfo>
      </VideoCoverContainer>
    )
  }
}

const Div: StyledFunction<CoverAssetProps> = styled.div
export const VideoCoverAsset = Div`
  background: url(${props => props.src || ""}) no-repeat center center;
  background-size: cover;
  background-color: black;
`

const VideoCoverOverlay = styled.div`
  opacity: 0.75;
  background-color: black;
`

const VideoCoverInfoRow = styled(MaxRow)`
  align-items: flex-end;
`

const VideoCoverInfo = styled.div`
  display: flex;
  align-items: flex-end;
  box-sizing: border-box;
  padding-bottom: 60px;
  z-index: 1;
  ${IconVideoPlay} {
    height: 60px;
    width: 44px;
    margin-right: 15px;
  }
  ${media.sm`
    padding-bottom: 40px;
  `}
`

const CoverDiv: StyledFunction<CoverProps> = styled.div
export const VideoCoverContainer = CoverDiv`
  position: relative;
  width: 100vw;
  height: 100vh;
  color: white;
  opacity: ${props => props.hideCover ? '0' : '1'};
  visibility: ${props => props.hideCover ? 'hidden' : 'visible'};
  transition: opacity 0.5s ease visibility 0.5s ease;
  ${VideoCoverAsset}, ${VideoCoverOverlay}, ${VideoCoverInfo} {
    position: absolute;
    width: 100%;
    height: 100%;
  }
`

const MediaDescription = styled.div`
  ${Fonts.garamond("s23")}
  ${media.sm`
    ${Fonts.garamond("s19")}
    margin-top: 30px;
  `}
`
