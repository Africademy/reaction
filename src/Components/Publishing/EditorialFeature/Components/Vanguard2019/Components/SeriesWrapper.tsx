import { Box, Sans } from "@artsy/palette"
import { ArticleData } from "Components/Publishing/Typings"
import React from "react"
import { VanguardArtistWrapper } from "./ArtistWrapper"

export const VanguardSeriesWrapper: React.SFC<{
  article: ArticleData
}> = props => {
  const { relatedArticles, title } = props.article
  return (
    <Box>
      <Sans size="6">Sub-Series Article: {title}</Sans>
      {/** map sub-series artist articles */}
      {relatedArticles &&
        relatedArticles.map(artistArticle => (
          <VanguardArtistWrapper article={artistArticle} />
        ))}
    </Box>
  )
}
