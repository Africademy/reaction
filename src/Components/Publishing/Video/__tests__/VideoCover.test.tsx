import { mount } from "enzyme"
import "jest-styled-components"
import React from "react"
import renderer from "react-test-renderer"
// import { VideoArticle } from "../../Fixtures/Articles"
import { Media } from "../../Fixtures/Components"
import { VideoCover } from "../VideoCover"

describe("Video Cover", () => {
  it("matches the snapshot", () => {
    const videoCover = renderer.create(
      <VideoCover
        media={Media[0]}
        seriesTitle="Future of Art"
        description="Lorem Ipsum Description"
      />
    ).toJSON()
    expect(videoCover).toMatchSnapshot()
  })

  // it("renders video info", () => {
  //   const component = mount(
  //     <VideoCover
  //       media={Media[0]}
  //       seriesTitle="Future of Art"
  //       description="Lorem Ipsum Description"
  //     />
  //   )
  //   // expect(toggleMute).toBeCalled()
  // })

  it("renders a description", () => {
    const component = mount(
      <VideoCover
        media={Media[0]}
        seriesTitle="Future of Art"
        description="Lorem Ipsum Description"
      />
    )
    expect(component.text()).toMatch("Lorem Ipsum Description")
  })
})
