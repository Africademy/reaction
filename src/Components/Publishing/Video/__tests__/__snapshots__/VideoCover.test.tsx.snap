// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Video Cover matches the snapshot 1`] = `
.c8 {
  width: 32px;
  height: 32px;
}

.c9 a {
  color: white;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c2 {
  background: url(https://artsy-vanity-files-production.s3.amazonaws.com/images/galerie-ceysson-benetiere_abmb.jpg) no-repeat center center;
  background-size: cover;
  background-color: black;
}

.c4 {
  background: linear-gradient(rgba(0,0,0,0.3),rgba(0,0,0,0.6));
}

.c6 {
  box-sizing: border-box;
  z-index: 1;
}

.c6 .c7 {
  height: 60px;
  width: 44px;
  cursor: pointer;
}

.c0 {
  position: absolute;
  top: 0;
  left: 0;
  width: 100vw;
  height: 100vh;
  color: white;
  opacity: 1;
  visibility: visible;
  -webkit-transition: opacity 0.25s ease,visibility 0.25s ease;
  transition: opacity 0.25s ease,visibility 0.25s ease;
}

.c0 .c1,
.c0 .c3,
.c0 .c5 {
  position: absolute;
  width: 100%;
  height: 100%;
}

<div
  className="c0"
>
  <div
    className="c1 c2"
    src="https://artsy-vanity-files-production.s3.amazonaws.com/images/galerie-ceysson-benetiere_abmb.jpg"
  />
  <div
    className="c3 c4"
  />
  <div
    className="c5 c6 sc-ifAKCX heICHz"
  >
    <div
      className="sc-kAzzGY jWPSYA"
    >
      <div
        className="sc-ifAKCX htzdbN"
      >
        <div
          className="sc-kAzzGY iQaGQD"
          onClick={[Function]}
          width="60px"
        >
          <svg
            className="c7 c8"
            version="1.1"
            viewBox="0 0 40 56"
            x="0px"
            xmlns="http://www.w3.org/2000/svg"
            y="0px"
          >
            <g
              fill="none"
              fillRule="evenodd"
              stroke="none"
              strokeWidth="1"
            >
              <polygon
                fill="white"
                points="0 0 0 56 39.3600006 28"
              />
            </g>
          </svg>
        </div>
        <div
          className="sc-kAzzGY eEGgS"
        >
          <div>
            <div
              className="sc-ifAKCX htzdbN"
            >
              <div
                className="c9 sc-bZQynM bOrPFB sc-EHOje girjXe"
                fontFamily="Unica77LLWebRegular, 'Helvetica Neue', Helvetica, Arial, sans-serif"
                fontSize="14px"
              >
                Future of Art
              </div>
              <div
                className="sc-bZQynM bOrPFB sc-EHOje CMaEi"
                fontFamily="Unica77LLWebRegular, 'Helvetica Neue', Helvetica, Arial, sans-serif"
                fontSize="14px"
              >
                02:28
              </div>
            </div>
            <div
              className="sc-bZQynM bOrPFB sc-EHOje BRZmD"
              fontFamily="Unica77LLWebRegular, 'Helvetica Neue', Helvetica, Arial, sans-serif"
              fontSize="42px"
            >
              Trevor Paglan
            </div>
          </div>
        </div>
      </div>
      <div
        className="sc-kAzzGY kccIrc"
      >
        <div
          className="sc-gzVnrw ibZhkv sc-EHOje evsrro"
          fontFamily="'Adobe Garamond W08', 'adobe-garamond-pro', 'AGaramondPro-Regular', 'Times New Roman', Times, serif"
          fontSize={
            Array [
              "18px",
              "22px",
              "22px",
              "22px",
            ]
          }
        >
          The elegant spiral of the Nautilus shell, the sinuous pattern of the banks of a river, or the swirling vortex street of clouds - patterns exist.
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Video Cover matches the snapshot with edit props 1`] = `
.c8 {
  width: 32px;
  height: 32px;
}

.c9 a {
  color: white;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c2 {
  background: url(https://artsy-vanity-files-production.s3.amazonaws.com/images/galerie-ceysson-benetiere_abmb.jpg) no-repeat center center;
  background-size: cover;
  background-color: black;
}

.c4 {
  background: linear-gradient(rgba(0,0,0,0.3),rgba(0,0,0,0.6));
}

.c6 {
  box-sizing: border-box;
  z-index: 1;
}

.c6 .c7 {
  height: 60px;
  width: 44px;
  cursor: pointer;
}

.c0 {
  position: absolute;
  top: 0;
  left: 0;
  width: 100vw;
  height: 100vh;
  color: white;
  opacity: 1;
  visibility: visible;
  -webkit-transition: opacity 0.25s ease,visibility 0.25s ease;
  transition: opacity 0.25s ease,visibility 0.25s ease;
}

.c0 .c1,
.c0 .c3,
.c0 .c5 {
  position: absolute;
  width: 100%;
  height: 100%;
}

<div
  className="c0"
>
  <div
    className="c1 c2"
    src="https://artsy-vanity-files-production.s3.amazonaws.com/images/galerie-ceysson-benetiere_abmb.jpg"
  />
  <div
    className="c3 c4"
  />
  <div
    className="c5 c6 sc-ifAKCX heICHz"
  >
    <div
      className="sc-kAzzGY jWPSYA"
    >
      <div
        className="sc-ifAKCX htzdbN"
      >
        <div
          className="sc-kAzzGY iQaGQD"
          onClick={[Function]}
          width="60px"
        >
          <svg
            className="c7 c8"
            version="1.1"
            viewBox="0 0 40 56"
            x="0px"
            xmlns="http://www.w3.org/2000/svg"
            y="0px"
          >
            <g
              fill="none"
              fillRule="evenodd"
              stroke="none"
              strokeWidth="1"
            >
              <polygon
                fill="white"
                points="0 0 0 56 39.3600006 28"
              />
            </g>
          </svg>
        </div>
        <div
          className="sc-kAzzGY eEGgS"
        >
          <div>
            <div
              className="sc-ifAKCX htzdbN"
            >
              <div
                className="c9 sc-bZQynM bOrPFB sc-EHOje girjXe"
                fontFamily="Unica77LLWebRegular, 'Helvetica Neue', Helvetica, Arial, sans-serif"
                fontSize="14px"
              >
                Future of Art
              </div>
              <div
                className="sc-bZQynM bOrPFB sc-EHOje CMaEi"
                fontFamily="Unica77LLWebRegular, 'Helvetica Neue', Helvetica, Arial, sans-serif"
                fontSize="14px"
              >
                02:28
              </div>
            </div>
            <div
              className="sc-bZQynM bOrPFB sc-EHOje BRZmD"
              fontFamily="Unica77LLWebRegular, 'Helvetica Neue', Helvetica, Arial, sans-serif"
              fontSize="42px"
            >
              <div>
                Child 
                title
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        className="sc-kAzzGY kccIrc"
      >
        <div
          className="sc-gzVnrw ibZhkv sc-EHOje evsrro"
          fontFamily="'Adobe Garamond W08', 'adobe-garamond-pro', 'AGaramondPro-Regular', 'Times New Roman', Times, serif"
          fontSize={
            Array [
              "18px",
              "22px",
              "22px",
              "22px",
            ]
          }
        >
          <div>
            Child 
            description
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
